array.c

```diff
@@ -6320,25 +6364,62 @@ ary_max_opt_string(VALUE ary, long i, VALUE vmax)
 
 /*
  *  call-seq:
- *     ary.max                     -> obj
- *     ary.max {|a, b| block}      -> obj
- *     ary.max(n)                  -> array
- *     ary.max(n) {|a, b| block}   -> array
+ *    array.max -> element
+ *    array.max {|a, b| ... } -> element
+ *    array.max(n) -> new_array
+ *    array.max(n) {|a, b| ... } -> new_array
  *
- *  Returns the object in _ary_ with the maximum value. The
- *  first form assumes all objects implement <code><=></code>;
- *  the second uses the block to return <em>a <=> b</em>.
+ *  Returns an element of +self+ or a new array of elements selected from +self+.
```

Comment:
I think the first line of the min and max documentation should say something about maximum/minimum values, as otherwise the start of the min and max documentation after the call-seq are exactly the same, and that is not good.  Otherwise, the changes are fine.